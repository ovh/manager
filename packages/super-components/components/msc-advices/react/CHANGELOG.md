# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.1.1](https://github.com/ovh/manager/compare/@ovhcloud/msc-react-advices@1.1.0...@ovhcloud/msc-react-advices@1.1.1) (2023-12-05)

**Note:** Version bump only for package @ovhcloud/msc-react-advices





# [1.1.0](https://github.com/ovh/manager/compare/@ovhcloud/msc-react-advices@1.0.0...@ovhcloud/msc-react-advices@1.1.0) (2023-12-05)


### Features

* msc-advices ([670db53](https://github.com/ovh/manager/commit/670db53549b433db305fa4e6875d804dc2da56c3))
