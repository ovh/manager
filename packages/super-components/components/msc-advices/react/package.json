{
  "name": "@ovhcloud/msc-react-advices",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for msc-advices super component",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/msc-advices#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/msc-advices/react"
  },
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "clean": "rimraf dist",
    "compile": "npm run compile:esm && npm run compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json",
    "msc:build": "npm run clean && npm run compile",
    "prepare": "yarn msc:build"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-advices": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "18.2.22",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "@ovhcloud/ods-components": "^16.0.1",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
