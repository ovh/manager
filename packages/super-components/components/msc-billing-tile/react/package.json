{
  "name": "@ovhcloud/msc-react-billing-tile",
  "version": "1.1.1",
  "private": true,
  "description": "React specific wrapper for Billing Tile super component",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/msc-billing-tile#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/msc-billing-tile/react"
  },
  "license": "BSD-3-Clause",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "yarn msc:build",
    "clean": "rimraf dist",
    "compile": "yarn compile:esm && yarn compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json",
    "msc:build": "yarn clean && yarn compile",
    "prepare": "yarn msc:build"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-billing-tile": "^1.1.1",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "18.2.22",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
