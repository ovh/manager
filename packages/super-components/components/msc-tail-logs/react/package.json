{
  "name": "@ovhcloud/msc-react-tail-logs",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for Tail Logs super components",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/msc-tail-logs#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/msc-tail-logs/react"
  },
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "yarn msc:build",
    "clean": "rimraf dist",
    "compile": "yarn compile:esm && yarn compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json",
    "msc:build": "yarn clean && yarn compile",
    "prepare": "yarn msc:build"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-tail-logs": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "18.2.22",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
