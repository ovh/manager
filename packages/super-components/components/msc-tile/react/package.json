{
  "name": "@ovhcloud/msc-react-tile",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for Manager super components",
  "keywords": [],
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "npm run clean && npm run compile",
    "clean": "rimraf dist",
    "compile": "npm run compile:esm && npm run compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-tile": "^1.0.0",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/react": "18.2.8",
    "@types/react-dom": "18.2.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "@ovhcloud/ods-stencil": "^13.0.1",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
