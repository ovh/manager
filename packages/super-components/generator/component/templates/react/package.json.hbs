{
  "name": "@ovhcloud/msc-react-{{ packageName }}",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for Manager super components",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/{{ componentTag }}#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/{{ componentTag }}/react"
  },
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "prepare": "yarn msc:build",
    "msc:build": "npm run clean && npm run compile",
    "clean": "rimraf dist",
    "compile": "npm run compile:esm && npm run compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-{{ packageName }}": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "17.0.39",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "@ovhcloud/ods-stencil": "^14.1.0",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
