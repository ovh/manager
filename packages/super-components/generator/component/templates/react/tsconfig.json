{
  "extends": "../../../tsconfig.react.json",
  "compilerOptions": {
    "outDir": "dist/esm",
    "declarationDir": "dist/types"
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"]
}
