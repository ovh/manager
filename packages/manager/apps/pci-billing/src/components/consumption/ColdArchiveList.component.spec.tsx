import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import ColdArchiveList from './ColdArchiveList.component';

describe('ColdArchiveList', () => {
  const mockColdArchives = [
    {
      name: 'archive',
      quantity: {
        value: 0,
        unit: 'Hour',
      },
      totalPrice: 0,
      region: 'RBX-ARCHIVE',
    },
    {
      name: 'restore',
      quantity: {
        value: 0,
        unit: 'Unit',
      },
      totalPrice: 0,
      region: 'RBX-ARCHIVE',
    },
    {
      name: 'archive-fees',
      quantity: {
        value: 0,
        unit: 'Hour',
      },
      totalPrice: 0,
      region: 'RBX-ARCHIVE',
    },
  ];

  it('matches snapshot with existing coldArchives', () => {
    const { asFragment } = render(
      <ColdArchiveList coldArchives={mockColdArchives} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with empty coldArchives', () => {
    const { asFragment } = render(<ColdArchiveList coldArchives={[]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
