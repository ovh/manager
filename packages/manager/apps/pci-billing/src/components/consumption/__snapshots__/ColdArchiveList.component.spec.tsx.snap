// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ColdArchiveList > matches snapshot with empty coldArchives 1`] = `
<DocumentFragment>
  <div
    class="my-3"
  >
    <osds-text
      class="mb-5"
      color="text"
      level="body"
      size="400"
    >
      pci_billing_cold_archive_no_entry
    </osds-text>
  </div>
</DocumentFragment>
`;

exports[`ColdArchiveList > matches snapshot with existing coldArchives 1`] = `
<DocumentFragment>
  <div
    class="my-3"
  >
    <div>
      <div
        class="contents px-[1px] "
      >
        <osds-table
          class="overflow-x-visible"
        >
          <!---->
          <table
            class="w-full border-collapse"
          >
            <thead>
              <tr>
                <th
                  class="text-left pl-4 h-11 whitespace-nowrap "
                >
                  <div
                    class=""
                    data-testid="header-region"
                  >
                    <span>
                      <osds-text
                        color="primary"
                        hue="800"
                        level="body"
                        size="500"
                      >
                        pci_billing_cold_archive_region
                      </osds-text>
                    </span>
                    <span
                      class="align-middle inline-block h-4 -mt-6"
                    >
                      <osds-icon
                        class="invisible"
                        color="primary"
                        name="arrow-down"
                        size="xxs"
                      />
                    </span>
                  </div>
                </th>
                <th
                  class="text-left pl-4 h-11 whitespace-nowrap "
                >
                  <div
                    class=""
                    data-testid="header-type"
                  >
                    <span>
                      <osds-text
                        color="primary"
                        hue="800"
                        level="body"
                        size="500"
                      >
                        pci_billing_cold_archive_type
                      </osds-text>
                    </span>
                    <span
                      class="align-middle inline-block h-4 -mt-6"
                    >
                      <osds-icon
                        class="invisible"
                        color="primary"
                        name="arrow-down"
                        size="xxs"
                      />
                    </span>
                  </div>
                </th>
                <th
                  class="text-left pl-4 h-11 whitespace-nowrap "
                >
                  <div
                    class=""
                    data-testid="header-consumption"
                  >
                    <span>
                      <osds-text
                        color="primary"
                        hue="800"
                        level="body"
                        size="500"
                      >
                        pci_billing_cold_archive_consumption
                      </osds-text>
                    </span>
                    <span
                      class="align-middle inline-block h-4 -mt-6"
                    >
                      <osds-icon
                        class="invisible"
                        color="primary"
                        name="arrow-down"
                        size="xxs"
                      />
                    </span>
                  </div>
                </th>
                <th
                  class="text-left pl-4 h-11 whitespace-nowrap "
                >
                  <div
                    class=""
                    data-testid="header-price"
                  >
                    <span>
                      <osds-text
                        color="primary"
                        hue="800"
                        level="body"
                        size="500"
                      >
                        pci_billing_cold_archive_price
                      </osds-text>
                    </span>
                    <span
                      class="align-middle inline-block h-4 -mt-6"
                    >
                      <osds-icon
                        class="invisible"
                        color="primary"
                        name="arrow-down"
                        size="xxs"
                      />
                    </span>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="border-solid border-[1px] h-[3.25rem] border-[var(--ods-color-blue-200)]"
              >
                <td
                  class="text-left pl-4"
                >
                  <osds-text
                    color="text"
                    level="body"
                  >
                    RBX-ARCHIVE
                  </osds-text>
                </td>
                <td
                  class="text-left pl-4"
                >
                  <div>
                    <osds-text
                      color="text"
                      level="body"
                    >
                      pci_billing_cold_archive_type_archive_label
                    </osds-text>
                  </div>
                </td>
                <td
                  class="text-left pl-4"
                >
                  <osds-text
                    color="text"
                    level="body"
                  >
                    pci_billing_cold_archive_consumption_value_gbih
                  </osds-text>
                </td>
                <td
                  class="text-left pl-4"
                >
                  <osds-text
                    color="text"
                    level="body"
                  >
                    0.00 undefined
                  </osds-text>
                </td>
              </tr>
              <tr
                class="border-solid border-[1px] h-[3.25rem] border-[var(--ods-color-blue-200)]"
              >
                <td
                  class="text-left pl-4"
                >
                  <osds-text
                    color="text"
                    level="body"
                  >
                    RBX-ARCHIVE
                  </osds-text>
                </td>
                <td
                  class="text-left pl-4"
                >
                  <div>
                    <osds-text
                      color="text"
                      level="body"
                    >
                      pci_billing_cold_archive_type_restore_label
                    </osds-text>
                  </div>
                </td>
                <td
                  class="text-left pl-4"
                >
                  <osds-text
                    color="text"
                    level="body"
                  >
                    pci_billing_cold_archive_consumption_value_gbi
                  </osds-text>
                </td>
                <td
                  class="text-left pl-4"
                >
                  <osds-text
                    color="text"
                    level="body"
                  >
                    0.00 undefined
                  </osds-text>
                </td>
              </tr>
              <tr
                class="border-solid border-[1px] h-[3.25rem] border-[var(--ods-color-blue-200)]"
              >
                <td
                  class="text-left pl-4"
                >
                  <osds-text
                    color="text"
                    level="body"
                  >
                    RBX-ARCHIVE
                  </osds-text>
                </td>
                <td
                  class="text-left pl-4"
                >
                  <div>
                    <osds-tooltip>
                      <osds-text
                        color="text"
                        level="body"
                      >
                        pci_billing_cold_archive_type_archive-fees_label
                      </osds-text>
                      <osds-tooltip-content
                        slot="tooltip-content"
                      >
                        <osds-text
                          color="text"
                          level="body"
                          size="100"
                        >
                          pci_billing_cold_archive_type_archive-fees_tooltip_1pci_billing_cold_archive_type_archive-fees_tooltip_2
                        </osds-text>
                      </osds-tooltip-content>
                    </osds-tooltip>
                  </div>
                </td>
                <td
                  class="text-left pl-4"
                >
                  <osds-text
                    color="text"
                    level="body"
                  >
                    pci_billing_cold_archive_consumption_value_gbih
                  </osds-text>
                </td>
                <td
                  class="text-left pl-4"
                >
                  <osds-text
                    color="text"
                    level="body"
                  >
                    0.00 undefined
                  </osds-text>
                </td>
              </tr>
            </tbody>
          </table>
        </osds-table>
      </div>
      <osds-pagination
        class="flex xs:justify-start md:justify-end"
        default-current-page="1"
        default-items-per-page="10"
        total-items="3"
        total-pages="1"
      >
        <span
          class="mr-3"
          slot="before-total-items"
        >
          common_pagination_of
        </span>
        <span
          class="ml-3"
          slot="after-total-items"
        >
          common_pagination_results
        </span>
      </osds-pagination>
    </div>
  </div>
</DocumentFragment>
`;
